### PR TITLE
feature: saga-remove spring

### PR DESCRIPTION
**Pull request by wt-better**
- [x] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

Seata saga current is strong dependency on spring. such as: 

1. code was dependency spring beans (InitializingBean ..etc)
2. expression evaluation dependency spring el 
3. spring bean service Task dependency spring ioc 

This is not conducive to saga's multilanguage building, so I has decouple saga with spring.

1. To ensure code compatibility, make new module named 'seata-saga-spring' and move related classes(not modified path) to this module.

2. 'seata-saga-tm' has spring dependency, this module is not necessary. First I removed this module and move class to 'seata-saga-spring' , I will sink TM's core capabilities ‘seata-saga-engine’ module later pr.

3. 'seata-saga-engine-store' remove the implementation of serialization and make it more pure

4. 'seata-saga-rm' has class named `StateMachineEngineHolder`, i moved to ‘seata-saga-engine’ module, and package name change, it`s not compatible. why ? I think it is inner use, users rarely use it.

### CODE REVIEW

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/impl/ProcessCtrlStateMachineEngine.java [hunk 1]
```diff
@@ -48,9 +43,15 @@
...
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.util.StringUtils;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/impl/ProcessCtrlStateMachineEngine.java]:
```suggestion
```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/impl/ProcessCtrlStateMachineEngine.java [hunk 2]
```diff
@@ -16,14 +16,9 @@
  */
 package org.apache.seata.saga.engine.impl;
 
-import java.util.Date;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-
 import org.apache.seata.common.exception.FrameworkErrorCode;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/impl/ProcessCtrlStateMachineEngine.java:25]:
```suggestion
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

import org.apache.seata.common.exception.FrameworkErrorCode;
```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/ServiceTaskStateHandler.java [hunk 3]
```diff
@@ -43,8 +39,11 @@
 import org.apache.seata.saga.statelang.domain.impl.ServiceTaskStateImpl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.context.ApplicationContextAware;
-import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/ServiceTaskStateHandler.java]:
```suggestion

```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/ServiceTaskStateHandler.java [hunk 4]
```diff
@@ -16,13 +16,9 @@
  */
 package org.apache.seata.saga.engine.pcext.handlers;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
 import org.apache.seata.common.exception.FrameworkErrorCode;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/ServiceTaskStateHandler.java:24]:
```suggestion
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.apache.seata.common.exception.FrameworkErrorCode;
```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/SubStateMachineHandler.java [hunk 5]
```diff
@@ -41,7 +37,11 @@
 import org.apache.seata.saga.statelang.domain.impl.SubStateMachineImpl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/SubStateMachineHandler.java]:
```suggestion

```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/SubStateMachineHandler.java [hunk 6]
```diff
@@ -16,12 +16,8 @@
  */
 package org.apache.seata.saga.engine.pcext.handlers;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
 import org.apache.seata.common.exception.FrameworkErrorCode;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/handlers/SubStateMachineHandler.java:24]:
```suggestion
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.apache.seata.common.exception.FrameworkErrorCode;
```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/interceptors/ServiceTaskHandlerInterceptor.java [hunk 7]
```diff
@@ -48,9 +44,14 @@
...
 import org.apache.seata.saga.statelang.domain.impl.StateInstanceImpl;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.util.StringUtils;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/interceptors/ServiceTaskHandlerInterceptor.java]:
```suggestion

```

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/interceptors/ServiceTaskHandlerInterceptor.java [hunk 8]
```diff
@@ -16,20 +16,16 @@
  */
 package org.apache.seata.saga.engine.pcext.interceptors;
 
-import java.util.Date;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-
 import org.apache.seata.common.exception.FrameworkErrorCode;
```
Comment (funky-eyes) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/interceptors/ServiceTaskHandlerInterceptor.java:24]:
```suggestion
import java.util.Date;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import org.apache.seata.common.exception.FrameworkErrorCode;
```

File: saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/config/DbStateMachineConfig.java [hunk 9]
```diff
@@ -16,34 +16,35 @@
...
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.DisposableBean;
 import org.springframework.util.StringUtils;
 
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
```
Comment (funky-eyes) [inline saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/config/DbStateMachineConfig.java]:
```suggestion
```

File: saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/config/DbStateMachineConfig.java [hunk 10]
```diff
@@ -16,34 +16,35 @@
  */
 package org.apache.seata.saga.engine.config;
 
-import javax.sql.DataSource;
-import java.sql.Connection;
-import java.sql.DatabaseMetaData;
-import java.sql.SQLException;
-
+import org.apache.seata.common.ConfigurationKeys;
```
Comment (funky-eyes) [inline saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/config/DbStateMachineConfig.java:24]:
```suggestion
import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.DatabaseMetaData;
import java.sql.SQLException;

import org.apache.seata.common.ConfigurationKeys;
```

File: saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/impl/DefaultStateMachineConfig.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.seata.saga.engine.invoker.impl.SpringBeanServiceInvoker;
+import org.apache.seata.saga.engine.utils.ResourceUtil;
+import org.apache.seata.saga.statelang.domain.DomainConstants;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.io.Resource;
+
+import java.io.IOException;
+import java.io.InputStream;
```
Comment (funky-eyes) [inline saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/impl/DefaultStateMachineConfig.java]:
```suggestion
```

File: saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/impl/DefaultStateMachineConfig.java [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.seata.saga.engine.impl;
+
+import org.apache.seata.saga.engine.config.AbstractStateMachineConfig;
```
Comment (funky-eyes) [inline saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/impl/DefaultStateMachineConfig.java:22]:
```suggestion
import java.io.IOException;
import java.io.InputStream;

import org.apache.seata.saga.engine.config.AbstractStateMachineConfig;
```

File: saga/seata-saga-statelang/src/test/java/org/apache/seata/saga/statelang/parser/StateParserTests.java [hunk 13]
```diff
@@ -16,29 +16,27 @@
...
 import org.apache.seata.saga.statelang.parser.utils.IOUtils;
+import org.apache.seata.saga.statelang.validator.ValidationException;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
-import org.springframework.core.io.ClassPathResource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+import java.util.Map;
```
Comment (funky-eyes) [inline saga/seata-saga-statelang/src/test/java/org/apache/seata/saga/statelang/parser/StateParserTests.java]:
```suggestion
```

File: saga/seata-saga-statelang/src/test/java/org/apache/seata/saga/statelang/parser/StateParserTests.java [hunk 14]
```diff
@@ -16,29 +16,27 @@
  */
 package org.apache.seata.saga.statelang.parser;
```
Comment (funky-eyes) [inline saga/seata-saga-statelang/src/test/java/org/apache/seata/saga/statelang/parser/StateParserTests.java:18]:
```suggestion

import java.io.IOException;
import java.io.InputStream;
import java.util.Date;
import java.util.Map;
```

**Review by funky-eyes — COMMENTED** [review]
code style need change

**Comment by wt-better** [comment]
> code style need change

i think not need change, history layout is not standard.

standard layout
image

seata-core  etc ..  other moudle also keep this layout

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**